common/admin/telemetry/telemetry_test.go

```diff
@@ -0,0 +1,125 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package telemetry
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/ceph/go-ceph/internal/admintest"
+)
+
+var radosConnector = admintest.NewConnector()
+
+type TelemetryReport struct {
+	DeviceReport json.RawMessage            `json:"device_report"`
+	Report       map[string]json.RawMessage `json:"report"`
+}
+
+func TestMgrAdmin_GetTelemetryStatus(t *testing.T) {
+	ra := radosConnector.Get(t)
+	admin := NewFromConn(ra)
+
+	tests := []struct {
+		name    string
+		want    *Status
+		wantErr bool
+	}{
+		{
+			name:    "happy Path",
+			want:    &Status{ChannelBasic: true, DeviceURL: "https://telemetry.ceph.com/device", Interval: 24},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := admin.Status(ra)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("MgrAdmin.GetTelemetryStatus() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !(got.ChannelBasic == tt.want.ChannelBasic && got.DeviceURL == tt.want.DeviceURL && got.Interval == tt.want.Interval) {
+				// if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("MgrAdmin.GetTelemetryStatus() = %+v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestMgrAdmin_GetTelemetryData(t *testing.T) {
+	ra := radosConnector.Get(t)
+	admin := NewFromConn(ra)
+
+	// a map container to decode the JSON structure into
+	c := TelemetryReport{
+		DeviceReport: []byte(""),
+		Report:       make(map[string]json.RawMessage),
+	}
+	tests := []struct {
+		// creative name of the test
+		name string
+		// keys we expect in the report
+		wantKeys map[string]interface{}
+		// if we expect this test to error
+		wantErr bool
+	}{
+		{
+			name: "happy Path",
+			wantKeys: map[string]interface{}{
+				"balancer":              true,
+				"channels":              true,
+				"channels_available":    true,
+				"collections_available": true,
+				"collections_opted_in":  true,
+				"config":                true,
+				"crashes":               true,
+				"created":               true,
+				"crush":                 true,
+				"fs":                    true,
+				"hosts":                 true,
+				"leaderboard":           true,
+				"license":               true,
+				"metadata":              true,
+				"mon":                   true,
+				"osd":                   true,
+				"pools":                 true,
+				"rbd":                   true,
+				"report_id":             true,
+				"report_timestamp":      true,
+				"report_version":        true,
+				"rgw":                   true,
+				"rook":                  true,
+				"services":              true,
+				"usage":                 true,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := admin.Data(ra)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("MgrAdmin.GetTelemetryData() error = %v, wantErr %v", err, tt.wantErr)
```

Comment:
```
=== RUN   TestMgrAdmin_GetTelemetryData/happy_Path
    telemetry_test.go:105: MgrAdmin.GetTelemetryData() error = rados: ret=-22, Invalid argument, wantErr false
--- FAIL: TestMgrAdmin_GetTelemetryData (0.05s)
    --- FAIL: TestMgrAdmin_GetTelemetryData/happy_Path (0.00s)
FAIL
```

This test failed to run. I'm not clear what happened. You can enable debugging of admin apis with the environment variable `GO_CEPH_TEST_DEBUG_TRACE=true` . If you want to use the CI you can even set the test script to (temporarily) enable it for all test runs.

Additionally, I recommend using [testify's](https://github.com/stretchr/testify) assert or require package, like most of the other tests in go-ceph use. It makes the error reporting in the tests simpler and the output more consistent.
(Note: I'm _not_ suggesting to remove the table style)

Reply:
I fixed the other two fsa.conn issues, maybe that broke the test runs? Let's try one more time. Any chance you can have the test run automatically when I push, so I don't have to wait for you?

Reply:
another possibility is that the telemetry manager module is not installed with all ceph versions in your test suite?

Reply:
if I go forward with option 2) then this would actually change the installation for the test run. IMO tests shouldn't change the system configuration, but I see the problem with 1) skipping the tests (I would actually prefer to run the test and expect an error).
Because of that I would prefer if you activate the modules in your test setups and I go forward with option 2) of skipping. If we skip, we should see the coverage drop and someone could investigate why if this ever becomes a problem. WDYT?

Reply:
Our test cluster is run only to support the test cases, while I would encourage you to restore the state of the cluster after the tests are complete I see no issue with making changes from within the tests suites.   We do this in other places.  The cephfs mirroring tests for example (cephfs/admin/mirror_workflow_test.go).
I don't really know how `you activate the modules in your test setups` would work because there's only one test setup and that is the containerized instance that's driven by the Makefile and scripts in this repo.  If it's a matter of handing off the work, I think #2 is perfectly acceptable as I or another one of the maintainers can make whatever follow up changes are needed in the future.